src/inference/src/dev/isync_infer_request.cpp

```diff
@@ -251,7 +252,10 @@ void ov::ISyncInferRequest::check_tensor(const ov::Output<const ov::Node>& port,
                     " expecting ",
                     port.get_shape(),
                     ".");
-    OPENVINO_ASSERT(tensor.is<ov::RemoteTensor>() || tensor.data() != nullptr, "Tensor data equal nullptr!");
+    // It seems tensor.is<ov::RemoteTensor>() is heavy function, it impacts performance greatly.
+    // OPENVINO_ASSERT(tensor.is<ov::RemoteTensor>() || tensor.data() != nullptr, "Tensor data equal nullptr!");
+    auto remote_tensor = std::dynamic_pointer_cast<ov::IRemoteTensor>(tensor._impl);
```

Comment:
Test result shows `tensor.is<ov::RemoteTensor>()` is the most factor to impact performance.

![image](https://github.com/openvinotoolkit/openvino/assets/31196718/40e6d495-88d2-4580-94fe-defe9f3cca87)